### ISSUE TITLE
onCreate() and onUpdate() entity events do not modify entity properties

### ISSUE THREAD

**Issue by DenisAnkh** [mikro-orm/mikro-orm#2804]
I am using micro-orm with nestjs. After update to 5.0.3 version I have noticed the following bug: entity properties that have `onCreate()` events are not updated (stay `undefined`) after saving. 

Here is detailed simplified example:

user.entity.ts:

```
class User {    
   @PrimaryKey({ length: 36, nullable: false, comment: 'UUID v4)' })
    id: string;

    @Property({ onCreate: () => new Date(), name: 'created_at', nullable: false, comment: 'DateTime in UTC format' })
    createdAt: Date;
}
```

client code:

```
const userEntity = new User();
userEntity.id = uuidv4();
await em.getRepository(User).persistAndFlush(userEntity);

console.log(userEntity.createdAt); // THE BUG IS HERE: it prints **undefined**, but it should print **the Date() autogenerated in onCreate()** event
```

Additional information: 

1) The problem has been appeared after update to 5.0.3 version. There has no the problem in  5.0.0, 5.0.1 and 5.0.2 version.
2) Executed SQL request is correct - it saves into the DB created_at Date value. So it seems the problem is that the property `createdAt` of `userEntity` variable is not updated after onCreate() defined in` @Property()`. 
3) I am using MySQL 8

**Comment by B4nan** [comment]
I can't reproduce this, are you sure its not just some context mismatch? Ideally provide executable repro.

In your example you have `userEntity` and then its called just `user`, that means its coming from a different request or is that just a typo? Tried both ways and it works fine.

**Comment by DenisAnkh** [comment]
@B4nan I have corrected my example above - it is `console.log(userEntity.createdAt);` not `console.log(user.createdAt);`. I just misspelled when described the issue. 

I will try to prepare executable example code to demonstrate the problem I've faced.
And the mikro-orm is awesome! :)

**Comment by B4nan** [comment]
Any updates?

**Comment by DenisAnkh** [comment]
@B4nan Last week I tested the code again (but after update mikro-orm to 5.0.4) and the problem has not raised anymore. I even rollback to 5.0.3 and retested the case. But the problem has not raised.  The issue is auto resolved.

So it seems it was a bug in VSCode debugger that displayed incorrect info after update the packages.  

As promised I have prepared the executable project that contains the code have described. Here is the link to it https://github.com/DenisAnkh/nestjs-clean-arch. Instructions of how to start it in the README file. The code in file https://github.com/DenisAnkh/nestjs-clean-arch/blob/main/src/user/adapters/data/user.mikro_orm.repository.ts in `save` method.
